test/dotcom_web/live/trip_planner_test.exs

```diff
@@ -20,19 +21,31 @@ defmodule DotcomWeb.Live.TripPlannerTest do
     stub_otp_results(itineraries)
   end
 
-  defp update_trip_id(itinerary, trip_id) do
-    updated_transit_leg =
-      itinerary.legs
-      |> Enum.at(1)
-      |> update_in([:trip, :gtfs_id], fn _ -> "ma_mbta_us:#{trip_id}" end)
-
-    itinerary
-    |> Map.update!(:legs, &List.replace_at(&1, 1, updated_transit_leg))
-  end
-
-  defp update_start_time(itinerary, start_time) do
-    itinerary
-    |> Map.put(:start, DateTime.new!(Date.utc_today(), start_time, "America/New_York"))
+  # For a list of headsigns, create a bunch of itineraries that would be grouped
+  # by the Trip Planner's logic
+  defp grouped_itineraries_from_headsigns([initial_headsign | _] = headsigns) do
+    # Only MBTA transit legs show the headsigns right now, so ensure the
+    # generated legs are MBTA-only
+    base_leg =
+      Factory.build(:transit_leg, %{
+        agency: Factory.build(:agency, %{name: "MBTA"}),
+        route:
+          Factory.build(:route, %{gtfs_id: "mbta-ma-us:internal", type: Faker.Util.pick(0..4)}),
+        trip:
+          Factory.build(:trip, %{
+            direction_id: Faker.Util.pick(["0", "1"]),
+            trip_headsign: initial_headsign
+          })
+      })
+
+    base_itinerary = Factory.build(:itinerary, legs: [base_leg])
+
+    Enum.reduce(headsigns, [], fn headsign, itineraries ->
+      leg = update_in(base_leg, [:trip, :trip_headsign], fn _ -> headsign end)
+      itinerary = update_in(base_itinerary, [:legs], fn _ -> [leg] end)
+      [itinerary | itineraries]
+    end)
+    |> Enum.reverse()
```

Comment:
```suggestion
    Enum.map(headsigns, fn headsign ->
      leg = update_in(base_leg, [:trip, :trip_headsign], fn _ -> headsign end)
      %{base_itinerary | legs: [leg]}
    end)
```

Per [this comment](https://github.com/mbta/dotcom/pull/2260/files#r1882440446), I'm not sure we need this logic at all, but if we do, then I think this version is equivalent?